0.23.0: CMake error when trying to build with tests disabled (-DBUILD_TESTS:BOOL=FALSE)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

Trying to build 0.23.0 (update our Exherbo Linux package from 0.22.2 to 0.23.0) with the newly introduced CMake variable for tests passing `-DBUILD_TESTS:BOOL=FALSE` results in a CMake configure error:

```
[...]
-- Tray icon disabled
-- Checking for module 'systemd'
--   Found systemd, version 255
CMake Error at cmake/targets/common.cmake:60 (set_source_files_properties):
  set_source_files_properties given non-existent DIRECTORY
  /var/tmp/paludis/build/net-remote-sunshine-0.23.0/work/sunshine-0.23.0/tests
Call Stack (most recent call first):
  CMakeLists.txt:56 (include)


CMake Error at cmake/targets/common.cmake:65 (set_source_files_properties):
  set_source_files_properties given non-existent DIRECTORY
  /var/tmp/paludis/build/net-remote-sunshine-0.23.0/work/sunshine-0.23.0/tests
Call Stack (most recent call first):
  CMakeLists.txt:56 (include)


CMake Error at cmake/targets/common.cmake:76 (set_source_files_properties):
  set_source_files_properties given non-existent DIRECTORY
  /var/tmp/paludis/build/net-remote-sunshine-0.23.0/work/sunshine-0.23.0/tests
Call Stack (most recent call first):
  CMakeLists.txt:56 (include)


-- Checking for module 'udev'
--   Found udev, version 255

-- Configuring incomplete, errors occurred!
```

Complete build log: [sunshine-0.23.0.log](https://github.com/LizardByte/Sunshine/files/14897193/sunshine-0.23.0.log)

### Expected Behavior

Do not fail to build when disabling building the tests.

### Additional Context

Passing `-DBUILD_TESTS:BOOL=TRUE` to have building the tests enabled makes CMake configure pass and the build of 0.23.0 succeed.

### Host Operating System

Linux

### Operating System Version

-

### Architecture

64 bit

### Sunshine commit or version

0.23.0

### Package

other (not listed)

### GPU Type

AMD

### GPU Model

RX 7800 XT

### GPU Driver/Mesa Version

mesa 24.04

### Capture Method (Linux Only)

PipeWire/X11/KMS

### Config

```Shell
-
```


### Apps

_No response_

### Relevant log output

```shell
-
```

--- a/cmake/targets/common.cmake
+++ b/cmake/targets/common.cmake
@@ -56,14 +56,20 @@ endif()
 
 # custom compile flags, must be after adding tests
 
+if (NOT BUILD_TESTS)
+    set(TEST_DIR "")
+else()
+    set(TEST_DIR "${CMAKE_SOURCE_DIR}/tests")
+endif()
+
 # src/upnp
 set_source_files_properties("${CMAKE_SOURCE_DIR}/src/upnp.cpp"
-        DIRECTORY "${CMAKE_SOURCE_DIR}" "${CMAKE_SOURCE_DIR}/tests"
+        DIRECTORY "${CMAKE_SOURCE_DIR}" "${TEST_DIR}"
         PROPERTIES COMPILE_FLAGS -Wno-pedantic)
 
 # third-party/nanors
 set_source_files_properties("${CMAKE_SOURCE_DIR}/third-party/nanors/rs.c"
-        DIRECTORY "${CMAKE_SOURCE_DIR}" "${CMAKE_SOURCE_DIR}/tests"
+        DIRECTORY "${CMAKE_SOURCE_DIR}" "${TEST_DIR}"
         PROPERTIES COMPILE_FLAGS "-include deps/obl/autoshim.h -ftree-vectorize")
 
 # third-party/ViGEmClient
@@ -74,7 +80,7 @@ string(APPEND VIGEM_COMPILE_FLAGS "-Wno-class-memaccess ")
 string(APPEND VIGEM_COMPILE_FLAGS "-Wno-unused-function ")
 string(APPEND VIGEM_COMPILE_FLAGS "-Wno-unused-variable ")
 set_source_files_properties("${CMAKE_SOURCE_DIR}/third-party/ViGEmClient/src/ViGEmClient.cpp"
-        DIRECTORY "${CMAKE_SOURCE_DIR}" "${CMAKE_SOURCE_DIR}/tests"
+        DIRECTORY "${CMAKE_SOURCE_DIR}" "${TEST_DIR}"
         PROPERTIES
         COMPILE_DEFINITIONS "UNICODE=1;ERROR_INVALID_DEVICE_OBJECT_PARAMETER=650"
         COMPILE_FLAGS ${VIGEM_COMPILE_FLAGS})

